sync2/semaphore_test.go

```diff
@@ -133,6 +133,12 @@ func (suite *SemaphoreSuite) TestLotsOfWaiters(t *C) {
 	for found := 0; found < 1000; found++ {
 		<-c
```

Comment:
That's funny, the second change makes the test fail. Inserting a tiny sleep helps over this induced racing condition. I am confused. Increment appears to be blocking.
```
    s.Increment(2000)
    time.Sleep(time.Duration(1)*time.Nanosecond)
```